### DISCUSSION TITLE
PDFs from Google Docs Fail to Process for RAG Due to Null Bytes

### DISCUSSION THREAD

**Discussion by Fakamoto** [danny-avila/LibreChat#2243] [Troubleshooting]
### What happened?

PDF documents exported from Google Docs and uploaded through the RAG API are not processed correctly for later use within the system. Although the upload process completes without immediate errors, these PDFs contain null byte (0x00) characters that prevent them from being properly loaded into the PostgreSQL database. This results in the system's inability to utilize these documents for RAG operations, despite their successful upload.

### Steps to Reproduce

1. Export a document from Google Docs as a PDF.
2. Upload this PDF through the RAG API interface.
3. Notice that, despite a successful upload, the document cannot be used for RAG operations due to an issue related to null byte characters.

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
A string literal cannot contain NUL (0x00) characters.
```

### Screenshots

<img width="822" alt="image (1)" src="https://github.com/danny-avila/LibreChat/assets/81929318/80acbc29-90e1-4f5a-bf99-76bb658e055e">

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Thanks, do you have rag_api logs for when this happens?

**Comment by danny-avila** [comment]
Cannot reproduce. Steps taken:

**1. ✅ Export a document from Google Docs as a PDF.** 

![image](https://github.com/danny-avila/LibreChat/assets/110412045/dca0b465-92f4-4898-8952-fb44b2d0af85)
![image](https://github.com/danny-avila/LibreChat/assets/110412045/73538fa8-932b-45eb-8371-c9ed9e8cd862)

**2. ✅ Upload this PDF through the RAG API interface.** 

![image](https://github.com/danny-avila/LibreChat/assets/110412045/d1e34e45-8483-4d17-99e0-4f65005440bf)

**3. ❌ Notice that, despite a successful upload, the document cannot be used for RAG operations due to an issue related to null byte characters.** 

![image](https://github.com/danny-avila/LibreChat/assets/110412045/0b6ef591-35f2-4641-917a-ddd835876fa6)

**Comment by Fakamoto** [comment]
Thanks for the quick response @danny-avila

I will try to get you a document to reproduce the bug, for now, I can provide you with rag_api logs:
`Mar 26 21:29:11 docker[2265]: 2024-03-26 21:29:11,963 - httpx - INFO - HTTP Request: POST https://oursite.openai.azure.com//openai/deployments/text-embedding-3-small/embeddings?api-version=2023-05-15 "HTTP/1.1 200 OK"
Mar 26 21:29:12 docker[2265]: A string literal cannot contain NUL (0x00) characters.
Mar 26 21:29:12 docker[2265]: INFO:     172.17.0.1:45184 - "POST /embed HTTP/1.1" 200 OK`

**Comment by Fakamoto** [comment]
@danny-avila Please try with this file, as it breaks our prod and stage instances of Librechat. Forgot to mention we use Azure OpenAI, although I don't know if its relevant.
[yua-aiqm-ptt (2024-03-04 16 50 GMT-5) - Transcript.pdf](https://github.com/danny-avila/LibreChat/files/14808550/yua-aiqm-ptt.2024-03-04.16.50.GMT-5.-.Transcript.pdf)
